core/src/main/java/de/danoeh/antennapod/core/feed/util/ImageResourceUtils.java

```diff
@@ -1,61 +1,33 @@
 package de.danoeh.antennapod.core.feed.util;
 
-import de.danoeh.antennapod.core.asynctask.ImageResource;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
 import de.danoeh.antennapod.core.feed.FeedItem;
-import de.danoeh.antennapod.core.feed.FeedMedia;
 import de.danoeh.antennapod.core.preferences.UserPreferences;
 
 /**
- * Utility class to use the appropriate image resource based on {@link UserPreferences}
+ * Utility class to use the appropriate image resource based on {@link UserPreferences}.
  */
 public final class ImageResourceUtils {
 
     private ImageResourceUtils() {
     }
 
-    /**
-     * returns the image location, does prefer the episode cover if available.
-     */
-    public static String getEpisodeImageLocation(ImageResource resource) {
-        return resource.getImageLocation();
-    }
-
-
-    /**
-     * returns the image location, does prefer the episode cover if available and enabled in settings.
-     */
-    public static String getEpisodeListImageLocation(ImageResource resource) {
-
+    @Nullable
+    public static String getImageLocation(@NonNull FeedItem feedItem) {
```

Comment:
Please keep the javadoc saying that it uses the user selected preference. I think using a method name that clarifies that would also be good (a bit like it was before - it had the word "list" in it to show that lists need to be handled separately).

Reply:
Yes, after starting from scratch I kept the original method names.